tests/ui/traits/next-solver/coroutine-higher-ranked-rigid.rs

```diff
@@ -0,0 +1,33 @@
+//@ compile-flags: -Znext-solver
+//@ edition: 2024
+//@ check-pass
+
+// Regression test for <https://github.com/rust-lang/trait-system-refactor-initiative/issues/177>.
+// Coroutines erase all free lifetimes from their interior types. In the future in `from_request`,
+// that means the `'r` lifetime in the type `<T as FromRequest<'r>>::Assoc`, which is in the
+// `dyn Future` type it's awaiting. Normalizing this associated type, with its free lifetimes
+// replaced, means proving `T: FromRequest<'!0>`, which definitely doesn't hold.
```

Comment:
by "definitely", i mean today. in the future (in order, for example, to fix the 12893892 other issues with async) we ideally will imply that `T: FromRequest<'!0>` holds for the specific erased lifetime we took from the coroutine. Or at least we would imply that `'!0 = 'r`, so the param-env bound applies.